docarray/array/mixins/find.py

```diff
@@ -124,14 +127,10 @@ def find(
 
         from ... import Document, DocumentArray
 
-        if isinstance(query, dict):
+        if isinstance(query, dict) and filter is None:
             return self._filter(query)
-        elif isinstance(query, (DocumentArray, Document)):
-
-            if isinstance(query, Document):
-                query = DocumentArray(query)
-
-            _query = query.embeddings
+        elif query is None and isinstance(filter, dict):
+            return self._filter(filter)
```

Comment:
isn't it weird that giving query or filter as input works the same?

Reply:
this aims to keep the compatibility